frontend/src/components/NavBar.js

```diff
@@ -107,10 +110,130 @@ class NavBar extends React.Component<{
             });
     }
 
+    search(keyword) {
+        const url = '/rest/interpret/wsearch';
+
+        $.ajax({
+            url: url,
+            dataType: 'json',
+            type: 'POST',
+            cache: false,
+            data: JSON.stringify({ keywords: [keyword] }),
+            contentType: 'application/json; charset=utf-8',
+            success: (data) => {
+                console.log(data);
+            },
+            error: (xhr, status, err) => {
+                console.error(xhr.responseText);
+            },
+        });
+    }
+
+    handleChange = (e, { value }) => this.setState({ value });
+
+    handleResultSelect = (e, { result }) => {
+        window.open('/worksheets/' + result.uuid, '_self');
+    };
+
+    initialState = { isLoading: false, results: [], value: '' };
+
+    resultRenderer = ({ title, description }) => [
+        <div key='content' className='content'>
+            {title && <div dangerouslySetInnerHTML={{ __html: title }} className='title'></div>}
```

Comment:
You should always call `DOMPurify.sanitize(title)` before using `dangerouslySetInnerHTML`, or you have an XSS security risk (see https://github.com/cure53/DOMPurify -- and the same goes with `description`)